enhancements/storage/csi-azure-patches.md

```diff
@@ -0,0 +1,217 @@
+fixes to Azure CSI drivers go into that dependency.  However, for
+older version of OCP, bumping that dependency results in too many code
+changes, increasing the risk of introducing new issues on the CSI
+drivers.  As a result, we need a way to address specific issues in
+`cloud-provider-azure` without risking the overall stability of the
+CSI drivers.
+
+## Motivation
+
+
+Some of the fixes for our Azure CSI drivers are implemented primarily
+in the `cloud-provider-azure` dependency.This means that, once issues
+are fixed in the dependency, we simply need to bump that dependency in
+our CSI driver fork to get the bug fix.
+
+This works well for recent versions of the CSI driver, however,
+updating this dependency in older versions of OCP typically pulls in a
+huge amount of code changes that are often unrelated to the fix being
+addressed.
+
+This is problematic because it increases the risk of introducing new
+and unknown issues to our CSI drivers. In addition to that, it makes
+it hard for QE to cover all the incoming changes in their test plan.
+
+So far we've addressed this problem by directly patching the vendor
+directory (see
+[this](https://github.com/openshift/azure-disk-csi-driver/pull/60) and
+[this](https://github.com/openshift/azure-disk-csi-driver/pull/62)),
+however, this is not sustainable in the long term.
+
+### User Stories
+
+* As an OpenShift engineer,I want to fix issues in Azure CSI drivers
+without merging additional code, so that I can be sure I am not
+introducing other issues while doing so.
+
+### Goals
+
+
+The goal of this enhancement is to document a strategy for updating
+the `cloud-provider-azure` dependency within our CSI drivers' forks in
+older versions of OCP.
+
+### Non-Goals
+
+We will not cover other dependencies or forks.
+
+## Proposal
+
+We propose to have branches in `openshift/cloud-provider-azure`
+repository that only contains fixes for the Azure CSI drivers. Those
+branches are based off on the exact version that CSI drivers rely
+on. Additional bug fixes will be added on top of it.
+
+### Workflow Description
+
+The workflow is better described with an example. Consider the
+following problem:
+
+1. QE discovers an issue with the `azure-disk-csi-driver` in OCP 4.12,
+   with a bugfix being available in a newer `cloud-provider-azure`
+   dependency.
+1. In OCP 4.12, the driver depends on the version `v0.7.0` of
+   `cloud-provider-azure`.
+1. A bugfix is available in `v0.11.0`.
+1. Bumping the dependency to that version brings in several unrelated
+   code changes.
+
+Our proposed solution:
+
+1. In `openshift/cloud-provider-azure`, a Staff Engineer manually
+   creates a branch called `csi-4.12-patches` based on the upstream
+   `v0.7.0`.
+   1. This only needs to be done once per OCP version, on demand.
+1. The engineer working on the issue then creates a local branch off
+   of `csi-4.12-patches` and then applies the fix for the given issue
+   on top of it.
```

Comment:
This will create a branch directly from upstream code. I think someone (the staff engineer?) will need to at least update OWNERS there.

Now the question is, who should be the owner? I'd suggest the storage team. The branch won't be used to build any cloud-provider-azure or its CCM, it will be used only as a library and vendored in our CSI drivers.